src/main.lua

```diff
@@ -2810,31 +2822,47 @@ addEventingHandler("releaseName", utils.hasMatchingTag("Action", ActionMap.Relea
 		return
 	end
 
+	-- Removes the record and primary names, then create the auction
+	---@param releasedName string
+	---@param releaseTimestamp number
+	---@param releasedInitiator string
+	local removeRecordsAndCreateAuction = function(releasedName, releaseTimestamp, releasedInitiator)
+		local removedRecord = arns.removeRecord(releasedName)
+		local removedPrimaryNamesAndOwners = primaryNames.removePrimaryNamesForBaseName(releasedName) -- NOTE: this should be empty if there are no primary names allowed before release
+		local auction = arns.createAuction(releasedName, releaseTimestamp, releasedInitiator)
+		return {
+			removedRecord = removedRecord,
+			removedPrimaryNamesAndOwners = removedPrimaryNamesAndOwners,
+			auction = auction,
+		}
+	end
+
 	-- we should be able to create the auction here
-	local status, auctionData = eventingPcall(msg.ioEvent, function(error)
+	local status, createAuctionDataOrError = eventingPcall(msg.ioEvent, function(error)
 		ao.send({
 			Target = msg.From,
 			Action = "Invalid-" .. ActionMap.ReleaseName .. "-Notice",
 			Error = "Auction-Creation-Error",
 			Data = tostring(error),
 		})
-		return
-	end, arns.createAuction, name, timestamp, initiator)
+	end, removeRecordsAndCreateAuction, name, timestamp, initiator)
 
-	if not status or not auctionData then
+	if not status or not createAuctionDataOrError then
 		ao.send({
 			Target = msg.From,
 			Action = "Invalid-" .. ActionMap.ReleaseName .. "-Notice",
 			Error = "Auction-Creation-Error",
-			Data = tostring(auctionData),
+			Data = tostring(createAuctionDataOrError),
 		})
 		return
 	end
 
 	-- add the auction result fields
 	addAuctionResultFields(msg.ioEvent, {
 		name = name,
-		auction = auctionData,
+		auction = createAuctionDataOrError.auction,
+		removedRecord = createAuctionDataOrError.removedRecord,
+		removedPrimaryNamesAndOwners = createAuctionDataOrError.removedPrimaryNamesAndOwners,
```

Comment:
Doesn't look as though `addAuctionResultFields` has been updated to make use of this yet. Is that intended?

Reply:
no, just missed

Reply:
Created a ticket